17 home filter styling

lucascumsille opened the pull request:
I have added two layout options for the filter. Let me know which one you prefer. I prefer the second version, which saves some vertical space, aligns the items to the left and doesn't give us the empty spacing below the search bar.

### Option 1
<img width="1424" alt="Screenshot 2022-09-22 at 12 33 11" src="https://user-images.githubusercontent.com/13790153/191903142-3acc3d9d-d0ae-4ab6-bfc6-38b299b8767d.png">

<img width="1424" alt="Screenshot 2022-09-22 at 12 33 06" src="https://user-images.githubusercontent.com/13790153/191903147-0ffc8d98-5011-4069-b290-7b099afd79ad.png">

### Option 2

<img width="1424" alt="Screenshot 2022-09-23 at 07 26 52" src="https://user-images.githubusercontent.com/13790153/191903242-350d2414-0c9b-4805-8d2c-4d079cb6365a.png">

A few notes:

- When I saw the left padding for the search bar, I thought it might be for an icon.
- At first, I didn't notice the "Advanced filters" button. I saw it when we were on yesterday's meeting with Alexander, so I thought to make it pop up more, but without giving the same priority as the "Submit" button.

### I have also included some minor but unrelated issues to this PR:

- I decreased the size of the text for the card title. I noticed they were a bit hard to read when the title was too long, with few characters/words in one line.
- I had a contrast issue between the primary colour and the grey background, so I darkened the version a bit.

### Preview of the cards titles
<img width="1424" alt="Screenshot 2022-09-23 at 07 32 44" src="https://user-images.githubusercontent.com/13790153/191903875-d4db95ad-12e4-49d1-8856-c270adbbd251.png">


They are separate commits, so let me know if you want me to get rid of any.

Fixes #17

lucascumsille committed:
Home page filter styling improvements

- New layout that takes up less space.
- Add search icon to council name input.
- Change to darker version of the primary colour, so it passes contrast
  against the gray background.
- Improve readability of tender list cards.
- Move search input into tender_filter.html, so that it can be nested
  inside a row along with other form controls. (This means the search
  also now appears on the /council/<slug> page, which will need
  testing out.)
- Limit width of sorting control.
- Give an explicit name to sumbit button.
- Re-indent tender_filter.html.

zarino commented on procurement/templates/procurement/home.html:
```diff
@@ -9,14 +9,16 @@
-            </fieldset>
+        <form action="" method="GET">
+            <div class="row">
+                <fieldset class="col-sm-auto me-auto">
```
A minor point, but seems odd to me that we use a `<fieldset>` here, since it’s wrapping just a single control. Usually you’d use a fieldset to group a set of related checkboxes or radios, or to create a logical "section" inside a form. I think, here, we’re really just after a `<div>`.

zarino replied:
One for @lucascumsille

zarino commented on procurement/templates/procurement/includes/tender_filter.html:
```diff
@@ -1,14 +1,14 @@
-            <fieldset class="col-sm-auto">
+            <fieldset class="d-flex align-items-end flex-wrap mb-md-2">
                 {% bootstrap_field filter.form.sort %}
                 {% bootstrap_field filter.form.state %}
```
As mentioned in Slack, my instinct is to hide this form control entirely, and always just return "completed" tenders. The aim of the site is to expose contracts that are expiring – that requires the contract to already have been awarded, which is [represented as a `tender.status` of `"complete"`](https://standard.open-contracting.org/1.1/en/schema/codelists/#tender-status).

zarino replied:
@alexander-griffen Would it be helpful to have a chat about this today?

alexander-griffen replied:
[This](https://github.com/mysociety/contract-countdown/pull/22/commits/b573028902d681fde9bac5506abd6ff99d9e8793) commit should solve this, by passing complete tenders only to the filter.

zarino commented on procurement/templates/procurement/includes/tender_filter.html:
```diff
@@ -1,14 +1,14 @@
 
-            <fieldset class="col-sm-auto">
+            <fieldset class="d-flex align-items-end flex-wrap mb-md-2">
                 {% bootstrap_field filter.form.sort %}
```
This `bootstrap_field` tag is currently outputting a `<select class="form-control">` when it should be a `<select class="form-select">` … how can we fix that?

zarino replied:
@alexander-griffen Would it be helpful to have a chat about this today?

alexander-griffen replied:
[This](https://github.com/mysociety/contract-countdown/pull/22/commits/d2bb85d332cd7535446ca3210e6c33172952c129) commit has solved this, by updating the bootstrap4 package to django-bootstrap5

zarino commented on procurement/templates/procurement/home.html:
```diff
@@ -9,14 +9,16 @@
+                    <label for="council-search" class="form-label">Search by postcode or council name</label>
+                    <div class="search-input">
+                        {% include 'procurement/icons/search.html' with classes='mr-2 flex-grow-0 flex-shrink-0' width='1.25em' height='1.25em' role='presentation' %}
+                        <input type="search" name="pc" id="council-search" class="form-control" value="{{ filter.form.pc.value }}">
```
`{{ filter.form.pc.value }}` here currently outputs `None` when a search hasn’t been performed yet. I guess we should probably use `{{ filter.form.pc.value|default_if_none:"" }}` to make sure the `value` attribute is empty when a search hasn’t been performed?

zarino replied:
One for @lucascumsille

zarino commented on procurement/static/css/_forms.scss:
```diff
@@ -0,0 +1,15 @@
+    }
+}
+
+.btn-dropdown {
```
Ah clever, making use of the browser’s default `<summary>` styling. I feel like, since this will only have the desired effect on a `<summary>` element, maybe it should use that, instead of a generic class like `.btn-dropdown`. So, `summary.btn` instead, maybe?

zarino replied:
I’d probably also add a comment to that line, explaining that `display: list-item` will reintroduce the browser’s default "toggle" triangle on the `summary` element. Because otherwise, seeing `display: list-item` on a button, with no other context, is a bit of a head scratcher!

zarino replied:
One for @lucascumsille

zarino commented on procurement/static/css/_forms.scss:
```diff
@@ -0,0 +1,15 @@
+}
+
+.form-check {
+    margin-bottom: 0.250rem;
```
I’d change this via the `$form-check-margin-bottom` variable instead.

zarino replied:
One for @lucascumsille

zarino reviewed (changes requested):
So lots of thoughts here, some of which might require intervention from @alexander-griffen or @struan.

If we drop the "State" form control, then I’d be tempted to put the "Search" and "Sort by" controls back onto a single line, with the submit button on the same line too. With the right flexbox layout, we can then hide the submit button if/when the form submission is handled by JavaScript instead, and the two remaining controls will still look ok.

alexander-griffen committed:
The list of tenders now will only contain 'completed' tenders.

I have removed the 'state' dropdown from the form, and also passed a
queryset which will, when evaluated, only contain 'complete' tenders.

A gotcha that I forsee with this implementation, is that this will need
to be passed to the view every time a FilterView is used (for example,
the email alert page.)

lucascumsille commented:
@zarino I have added a --fixup that fixes the problem listed above.

alexander-griffen committed:
Bootstrap4 updated to sjango-bootstrap5

This solves incorrect bootstrap_field tags being outputted.

zarino committed:
Disable server-side validation styles for Bootstrap form elements
